server/pulp/server/managers/schedule/repo.py

```diff
@@ -178,22 +175,20 @@ def create(cls, repo_id, distributor_id, publish_options, schedule,
         :return:    new schedule instance
         :rtype:     pulp.server.db.model.dispatch.ScheduledCall
         """
-        # validate the input
-        cls.validate_distributor(repo_id, distributor_id)
+        dist = model.Distributor.objects.get_or_404(repo_id=repo_id, distributor_id=distributor_id)
         utils.validate_keys(publish_options, _PUBLISH_OPTION_KEYS)
         utils.validate_initial_schedule_options(schedule, failure_threshold, enabled)
 
         task = repo_controller.queue_publish.name
         args = [repo_id, distributor_id]
         kwargs = {'overrides': publish_options['override_config']}
-        resource = RepoDistributor.build_resource_tag(repo_id, distributor_id)
         schedule = ScheduledCall(schedule, task, args=args, kwargs=kwargs,
-                                 resource=resource, failure_threshold=failure_threshold,
+                                 resource=dist.resource_tag, failure_threshold=failure_threshold,
                                  enabled=enabled)
         schedule.save()
 
         try:
-            cls.validate_distributor(repo_id, distributor_id)
+            model.Distributor.objects.get_or_404(repo_id=repo_id, distributor_id=distributor_id)
```

Comment:
For a future effort, it seems like it might be helpful to have a method ``objects.exists_or_404`` that does the cheapest operation possible to determine if the thing exists, without actually retrieving it.